### PR TITLE
feat(workspaces): folders

### PR DESCRIPTION
**Pull request by foochifa**
## Problem

Closes FRM-1245

## Solution

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [ ] No - this PR is backwards compatible  

## Tests

QA test in notion

### CODE REVIEW

File: frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx [hunk 1]
```diff
@@ -40,11 +52,19 @@ const Template: Story<CreateFormModalProps> = (args) => {
...
+      currentWorkspace={MOCK_DEFAULT_WORKSPACE._id}
+      defaultWorkspace={MOCK_DEFAULT_WORKSPACE}
+      setCurrentWorkspace={() => {
+        return
+      }}
+    >
+      <CreateFormModal
+        {...args}
+        {...modalProps}
+        onClose={() => console.log('close modal')}
```
Comment (tshuli) [inline frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx:65]: eh why need the close modal log 😅

File: frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx [hunk 2]
```diff
@@ -82,13 +102,21 @@ export const StorageModeAckScreen = () => {
...
-      </ModalContent>
-    </Modal>
+    <WorkspaceProvider
+      currentWorkspace={MOCK_DEFAULT_WORKSPACE._id}
+      defaultWorkspace={MOCK_DEFAULT_WORKSPACE}
+      setCurrentWorkspace={() => {
+        return
+      }}
+    >
+      <Modal isOpen onClose={() => console.log('close modal')} size="full">
```
Comment (tshuli) [inline frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx:112]: same - console log

File: frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx (hunk 1 above)
Comment (foochifa) [reply frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx:65]: Oh this is in prod actually, its in our storybook, not sure why there is console log too 😅 https://github.com/opengovsg/FormSG/blob/develop/frontend/src/features/workspace/components/CreateFormModal/CreateFormModal.stories.tsx

File: src/app/modules/workspace/workspace.controller.ts [hunk 3]
```diff
@@ -0,0 +1,298 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { AuthedSessionData } from 'express-session'
+import { StatusCodes } from 'http-status-codes'
+import { ErrorDto } from 'shared/types'
+import { WorkspaceDto } from 'shared/types/workspace'
+
+import {
+  WORKSPACE_MAX_TITLE_LENGTH,
+  WORKSPACE_MIN_TITLE_LENGTH,
+} from '../../../../shared/constants'
```
Comment (tshuli) [inline src/app/modules/workspace/workspace.controller.ts:10]: should this be an absolute path instead? 'shared/types/workspace'

Comment (foochifa) [reply src/app/modules/workspace/workspace.controller.ts:10]: I realised in the BE for some reason my build will fail if i use absolute path 😩 

Hence I think other files also use relative like `src/app/modules/submission/receiver/receiver.service.ts`

File: src/app/models/workspace.server.model.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import { ClientSession, Mongoose, Schema } from 'mongoose'
+
+import {
+  WORKSPACE_MAX_TITLE_LENGTH,
+  WORKSPACE_MIN_TITLE_LENGTH,
+} from '../../../shared/constants'
```
Comment (tshuli) [inline src/app/models/workspace.server.model.ts:6]: /suggestion - can use absolute path for shared

File: src/app/models/form.server.model.ts [hunk 5]
```diff
@@ -1140,6 +1139,17 @@ const compileFormModel = (db: Mongoose): IFormModel => {
     ).exec()
   }
 
+  FormSchema.statics.archiveForms = async function (
+    formIds: IFormSchema['_id'],
```
Comment (tshuli) [inline src/app/models/form.server.model.ts]: typing should be
`formIds: IFormSchema['_id'][]``

Comment (foochifa) [reply src/app/models/form.server.model.ts]: Thanks for the catch! [05cf6b9](https://github.com/opengovsg/FormSG/pull/6639/commits/05cf6b93ac3cf905fa3e9e86f8ef92b8ad4faba0)

**Review by tshuli — APPROVED** [review]

**Pull request merged**